firmware/src/ledger/signer/src/main.c

```diff
@@ -210,6 +210,34 @@ unsigned char io_event(unsigned char channel) {
     return 1;
 }
 
+static bool do_io_exchange(volatile unsigned int *rtx) {
+    BEGIN_TRY {
+        TRY {
+            *rtx = io_exchange(CHANNEL_APDU, *rtx);
+            return true;
+        }
+        CATCH_OTHER(e) {
+            *rtx = 0;
+            G_io_apdu_buffer[(*rtx)++] = 0x68;
+            G_io_apdu_buffer[(*rtx)++] = e & 0xFF;
+            return false;
+        }
+        FINALLY {
+        }
+    }
+    END_TRY;
+}
+
+static void main_loop() {
+    volatile unsigned int rtx = 0;
+
+    while (!hsm_exit_requested()) {
+        if (!do_io_exchange(&rtx))
+            continue;
```

Comment:
Aren't we ignoring a potential exception that could happen in `do_io_exchange` here?

Reply:
I don't quite follow. Ignoring it how?

Reply:
Oh, you're right. I thought the exception would be handled by `hsm_process_apdu`, which we're skipping, but it will be sent to the host in the next `do_io_exchange` call.

All good, never mind. I'll approve the PR, since my other comment is just an optional suggestion.

Reply:
Oh, gotcha. Thanks for clarifying!